### ISSUE TITLE
Add support for HTML5 number input type

### ISSUE THREAD

**Issue by vasilp** [customd/jquery-number#88]
As per specification input tag with type="number" are meant to be for numeric values.
I have tested the code with modified check **"if(this.is('input:text'))"**  to be **"if(this.is('input:text, [type="number"]'))"**, but seems like the number input type causes some side effects and the length cannot grow more than 4 symbols.

My update was here:
> // If we've specified to take the number from the target element,
> // we loop over the collection, and get the number.
> if( number === true )
> {
> 	// If this element is a number, then we add a keyup
> 	**if( this.is('input:text') )**
> 	{ .... }
> }

It will be good if there is a support for the new HTML5 input type number.

**Comment by Mihai-P** [comment]
I do not believe this can work for number fields. There are several problems the biggest one is that 12,000.00 is not a valid number fields, so the HTML5 field will show an error right away that the format is incorrect. Even if you disable that error the form will not submit that field (it will be empty value). So this cannot work for the html5 number fields.

**Comment by samatcd** [comment]
That's correct, It's impossible to have thousands separators working with number fields.

**Comment by samatcd** [comment]
The only possible way to do this would be to implement a hidden shadow field of 'number' type (which should trigger html5 input options) and use a cloned-in-place text input field to display the formatted number... although this might have some layout and form submission issues, but is a potential way forward.

**Comment by Mihai-P** [comment]
I was thinking the same thing, but it will get quite complicated to do this switcharoo I believe.

**Comment by tilparkmustafa** [comment]
I solved this problem

```
$(document).ready(function() {
			$(".money").keypress(function () {
				if($(this).attr('type') == 'tel') {
					$(this).attr('data-type', 'tel');
					if($(this).is(':focus')) {
						$(this).attr('type', 'text');
					} 
				}
			});

			$(".money").focusout(function () {
				if($(this).attr('data-type') == 'tel') {
					if(!$(this).is(':focus')) {
						$(this).removeAttr('data-type');
						$(this).attr('type', 'tel');
					}
				}
				
			});
		});
```

**Comment by RichardDavies** [comment]
A number formatting plugin needs to work with number fields so I'm a little confused as to why this issue hasn't been fixed yet especially given that the primary problem mentioned is that HTML5 number fields don't allow a thousands separator. Why not just disable the thousands separator for number fields?
